docs/authoring-components.md

```diff
@@ -478,9 +472,10 @@ A typical component within our codebase looks like so:
 packages
   ├── mdc-component
       ├── README.md # The component's README
-      ├── constants.js # The component's cssClasses/strings/numbers constants
-      ├── foundation.js # The component's foundation class
-      ├── index.js # The file that contains the vanilla component class, as well as exports the foundation
+      ├── adapter.ts # The adapter interface implemented by the component
+      ├── constants.ts # The component's cssClasses/strings/numbers constants
+      ├── foundation.ts # The component's foundation class
+      ├── index.ts # The file that contains the vanilla component class, as well as exports the foundation
```

Comment:
Add `component.ts` and update this comment to reflect that `index.ts` only re-exports the other files.